config/7.2.0/securebanking/ig/scripts/groovy/SignEventsResponse.groovy

```diff
@@ -0,0 +1,101 @@
+    }
+}
+
+Error response example:
+{
+    "Code": "500",
+    "Id": "ede36b552dc951d9836a127f16a7c033",
+    "Message": "[Status: 500 Internal Server Error]",
+    "Errors": [
+        {
+            "ErrorCode": "UK.OBIE.UnexpectedError",
+            "Message": "Internal error [Error signing event set payload, Cause: secretsProvider must be supplied]"
+        }
+    ]
+}
+
+Validate signature
+- You can validate the JWT SET signature in: https://jwt.davetonge.co.uk/
+- Copy a JWT 'set' value and paste in the web page.
+- JWKs URL endpoint to validate the signature: https://keystore.openbankingtest.org.uk/0015800001041REAAY/0015800001041REAAY.jwks
+ */
+
+def fapiInteractionId = request.getHeaders().getFirst("x-fapi-interaction-id");
+if (fapiInteractionId == null) fapiInteractionId = "No x-fapi-interaction-id"
+SCRIPT_NAME = "[SignEventsResponse] (" + fapiInteractionId + ") - "
+
+Map<String, Object> critClaims = new HashMap<>()
+critClaims.put("http://openbanking.org.uk/iat", System.currentTimeMillis() / 1000)
+critClaims.put("http://openbanking.org.uk/iss", aspspOrgId)
+critClaims.put("http://openbanking.org.uk/tan", "openbanking.org.uk")
+
+next.handle(context, request).thenOnResult({ response ->
```

Comment:
This is an optional comment, I think it would be nicer if this Filter were written in Java rather than Groovy.

Reply:
I have created a `DefaultSapiJwsSigner` that implements `SapiJwsSigner`. This new implementation of signer has been added to the Heap objects to simplify the Script filter.
**Heap**
```json
{
      "name": "DefaultSapiJwsSigner-RSASSA-PSS",
      "type": "com.forgerock.sapi.gateway.sign.DefaultSapiJwsSigner",
      "config": {
        "algorithm": "PS256",
        "signingKeyId": "jwt.signer",
        "kid": "&{ig.ob.aspsp.signing.kid}",
        "secretsProvider": "SecretsProvider-ASPSP"
      }
    }
```
**Filter**
```json
        {
          "comment": "Sign events from the RS response",
          "name": "SignEvents",
          "type": "ScriptableFilter",
          "config": {
            "type": "application/x-groovy",
            "file": "SignEventsResponse.groovy",
            "args": {
              "signer": "${heap['DefaultSapiJwsSigner-RSASSA-PSS']}",
              "aspspOrgId": "&{ob.aspsp.org.id}"
            }
          }
        }
```